oauth2l/__init__.py

```diff
@@ -256,6 +263,23 @@ def _GetCredentialsVia3LO(client_info, credentials_filename=None):
             credentials = None
     return credentials
 
+def _GetCredentialForServiceAccount(json_keyfile, scopes, 
```

Comment:
Can we share the code with above? I would expect a generic cache like this:

CacheEntry LookupCache(filename, client_id, scopes)

3LO client id and 2LO client id won't conflict, so there is no need to use 3LO and 2LO prefix.